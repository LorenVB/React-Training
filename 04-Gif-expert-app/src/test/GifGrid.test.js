const { shallow } = require('enzyme');
const { GifGrid } = require('../components/GifGrid');
const { useFetchGifs } = require('../hooks/useFetchGifs');
jest.mock('../hooks/useFetchGifs');

describe('Testing GifGrid component', () => {
  const category = 'Boku';

  it('should display component match', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should show items when images loaded by useFetchGifs', () => {

    const gifs = [{
        id:'ABC',
        url: 'https://localhost//cosa.jpg',
        title: 'Cualquier cosa'
    }]

    useFetchGifs.mockReturnValue({
        data: gifs,
        loading: false,
      });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        
  });

  });

