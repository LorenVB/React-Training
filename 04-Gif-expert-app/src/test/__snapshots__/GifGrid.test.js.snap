// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid component should display component match 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Boku
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid animate__animated animate__rubberBand"
  />
</Fragment>
`;

exports[`Testing GifGrid component should show items when images loaded by useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Boku
  </h3>
  <div
    className="card-grid animate__animated animate__rubberBand"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost//cosa.jpg"
    />
  </div>
</Fragment>
`;
